src/main/java/com/autotune/analyzer/recommendations/updater/vpa/VpaUpdaterImpl.java

```diff
@@ -0,0 +1,377 @@
+                throw new Exception(AnalyzerErrorConstants.RecommendationUpdaterErrors.INVALID_VPA_NAME);
+            } else {
+                LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.CHECKING_IF_VPA_PRESENT, vpaName));
+                NamespacedVerticalPodAutoscalerClient client = new DefaultVerticalPodAutoscalerClient();
+                VerticalPodAutoscalerList vpas = client.v1().verticalpodautoscalers().inAnyNamespace().list();
+
+                if (null != vpas && null != vpas.getItems() && !vpas.getItems().isEmpty()) {
+                    // TODO:// later we can also check here is the recommender is Kruize to confirm
+                    for (VerticalPodAutoscaler vpa : vpas.getItems()) {
+                        if (vpaName.equals(vpa.getMetadata().getName())) {
+                            LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.VPA_WITH_NAME_FOUND, vpaName));
+                            return true;
+                        }
+                    }
+                }
+                LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.VPA_WITH_NAME_NOT_FOUND, vpaName));
+                return false;
+            }
+        } catch (Exception e) {
+            LOGGER.error("Error while checking VPA presence: " + e.getMessage(), e);
+            return false;
+        }
+    }
+
+
+    /**
+     * Returns the VPA Object if present with the name
+     *
+     * @param vpaName String containing the name of the VPA object to search for
+     * @return VerticalPodAutoscaler if the VPA object with the specified name is present, null otherwise
+     */
+    private VerticalPodAutoscaler getVpaIsPresent(String vpaName) {
+        try {
+            if (null == vpaName || vpaName.isEmpty()) {
+                throw new Exception(AnalyzerErrorConstants.RecommendationUpdaterErrors.INVALID_VPA_NAME);
+            } else {
+                LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.CHECKING_IF_VPA_PRESENT, vpaName));
+                NamespacedVerticalPodAutoscalerClient client = new DefaultVerticalPodAutoscalerClient();
+                VerticalPodAutoscalerList vpas = client.v1().verticalpodautoscalers().inAnyNamespace().list();
+
+                if (null != vpas && null != vpas.getItems() && !vpas.getItems().isEmpty()) {
+                    for (VerticalPodAutoscaler vpa : vpas.getItems()) {
+                        if (vpaName.equals(vpa.getMetadata().getName())) {
+                            LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.VPA_WITH_NAME_FOUND, vpaName));
+                            return vpa;
+                        }
+                    }
+                }
+                LOGGER.info(String.format(AnalyzerConstants.RecommendationUpdaterConstants.InfoMsgs.VPA_WITH_NAME_NOT_FOUND, vpaName));
```

Comment:
This is called while applying recommendations right? Please use LOGGER.debug everywhere

Reply:
Yes, as we discussed, it will verify before applying the recommendations or before creating the vpa objects.